### ISSUE TITLE
3.8.4: quarkus.analytics.disabled ignored + command line answer not processed

### ISSUE THREAD

**Issue by rupert-jung-mw** [quarkusio/quarkus#41387]
Seems like `quarkus.analytics.disabled` just does nothing.

`quarkus.analytics.disabled` is set to false (both in application.yaml and application-dev-yaml):

![image](https://github.com/quarkusio/quarkus/assets/160733883/ffc97d03-8c60-4a30-a93a-e43c28d25ecf)

Config value itself is correctly set:
![image](https://github.com/quarkusio/quarkus/assets/160733883/36417bfe-4a66-4b6d-b38a-5fad93b26cd5)

... but ignored by Quarkus:
![image](https://github.com/quarkusio/quarkus/assets/160733883/c6326a9a-9b29-4f34-9506-a19cfb7ecb2e)

Even trying to answer the question via the console is not working, the input is not scanned:

![image](https://github.com/quarkusio/quarkus/assets/160733883/a01f5007-1cd8-446a-9d33-2a1d07e4a966)

_Originally posted by @rupert-jung-mw in https://github.com/quarkusio/quarkus/issues/34626#issuecomment-2185933766_

**Comment by maxandersen** [comment]
as documented on https://quarkus.io/usage/ it needs to be system property.

It is not a quarkus application property as its needed before quarkus is in any way involved.

**Comment by rupert-jung-mw** [comment]
Thank you @maxandersen ,

this has indeed been a missunderstanding on my side, thank you. Setting an env VAR named 'quarkus.analytics.disabled' with value 'true' made the message and delay disappear. But why was the 'n' not recognized in the command line (I am using IntelliJ with default settings?)

**Comment by rupert-jung-mw** [comment]
yes, see screenshot (newline after 'n'):

![image](https://github.com/quarkusio/quarkus/assets/160733883/85bc6107-1c60-477f-a400-8978a544f659)

**Comment by maxandersen** [comment]
ok, probably question for @brunobat  - are you simply opening the main class and run it using intellij via Run or some other mechanism?

**Comment by rupert-jung-mw** [comment]
Hello @maxandersen,

yes, I am just using the default main method automatically found by IntelliJ IDEA:

![image](https://github.com/quarkusio/quarkus/assets/160733883/9ec696c9-e903-4c91-b6ee-c61d8f7ee604)

**Comment by brunobat** [comment]
> Hello @maxandersen,
> 
> yes, I am just using the default main method automatically found by IntelliJ IDEA:
> 

I don't believe that execution method accepts any input.
Please execute on the command line the dev mode with `mvn quarkus:dev`.

**Comment by rupert-jung-mw** [comment]
Hi @brunobat,

I am using gradle, so I can not use any mvn functionality. But the execution method can in fact be configured in IntelliJ:
![image](https://github.com/quarkusio/quarkus/assets/160733883/26f2408d-ce6b-4217-84c7-710a018b1b06)

**Comment by brunobat** [comment]
@rupert-jung-mw the equivalent command to run Quarkus in dev mode with gradle is:
`./gradlew --console=plain quarkusDev`
For reference: https://quarkus.io/guides/gradle-tooling#dev-mode

**Comment by rupert-jung-mw** [comment]
@brunobat Thank you, I'll try that. So does this mean that Quarkus usage statistics command line scanning is just not expected to run via the default run configuration in IntelliJ? 

I worked around my using your suggested start mode with gradle + quarkusDev instead. Also used the environment variable to work around the problem with the usage statistics startup delay. So long it worked. :)

**Comment by brunobat** [comment]
Yes, @rupert-jung-mw and that question will only happen once per user... No need to answer yes or no all the time. 
Please see the details here: https://quarkus.io/usage/
Can we close this issue?

**Comment by rupert-jung-mw** [comment]
Hi @brunobat,

problem is, that it seems not possible to answer the question via the command line when just starting the main class because the command line scanning seems not working in the console (see screenshot).

**Comment by brunobat** [comment]
What console are you using? What operating system and version?

**Comment by rupert-jung-mw** [comment]
It's just the built-in console of IntelliJ IDEA 2024.1.2 . Windows 11x64 23H2.

**Comment by rodcheater** [comment]
FWIW this kinda ruins the out-of-the-box developer experience (clone repo, press run in the IDE). I’m considering whether I should add instructions to write to `~/.redhat/io.quarkus.analytics.localconfig` in our README but I probably shouldn’t have to do that.

**Comment by brunobat** [comment]
@rodcheater are you also on Windows?

**Comment by rodcheater** [comment]
> [@rodcheater](https://github.com/rodcheater) are you also on Windows?

No, not on Windows.

I’m talking about that the only way to get rid of the prompt or the 10 second delay is via a system property. I can’t use `application.properties`, therefore I can’t add something to version control, therefore every one of my teammates who goes to develop for our application must figure out on their own how to get rid of the prompt and 10 second delay. Not being able to answer the prompt when running in the IDE exacerbates the problem.

**Comment by rodcheater** [comment]
By the way, using `-Dquarkus.analytics.disabled=false` on the command line doesn’t seem to get rid of the prompt. Only `-Dquarkus.analytics.disabled=true` does.

**Comment by brunobat** [comment]
We need to schedule this for fixing but in the meantime, can't you just run the app in dev mode one and answer yes?... It will be valid for all projects. Just run `mvn quarkus:dev`.

**Comment by brunobat** [comment]
@rodcheater, @rupert-jung-mw. I took another look at this.
You mention that Only `-Dquarkus.analytics.disabled=true` gets rid of the prompt. That's on purpose.

Also, this is not supposed to be overridable per organisation by setting the prop in a repo, either in `application.properties` or `application-dev-yaml`
The question is personal, meaning per individual user and happens once, on the **first time the user runs dev mode** in any Quarkus project.

I don't think this is a bug because the behaviour matches the original intent of the feature.

Closing as won't do.
